src/components/structures/MatrixChat.tsx

```diff
@@ -112,117 +113,136 @@ const ONBOARDING_FLOW_STARTERS = [
+interface IScreen {
+    screen: string;
+    params?: object;
+}
+
+interface IRoomInfo {
+    room_id?: string;
+    room_alias?: string;
+    event_id?: string;
+
+    auto_join?: boolean;
+    highlighted?: boolean;
+    third_party_invite?: object;
+    oob_data?: object;
+    via_servers?: string[];
+}
+
+interface IProps { // TODO type things better
+    config: Record<string, any>;
+    serverConfig?: ValidatedServerConfig;
+    ConferenceHandler?: any;
+    onNewScreen: (string) => void;
+    enableGuest?: boolean;
+    // the queryParams extracted from the [real] query-string of the URI
+    realQueryParams?: Record<string, string>;
+    // the initial queryParams extracted from the hash-fragment of the URI
+    startingFragmentQueryParams?: Record<string, string>;
+    // called when we have completed a token login
+    onTokenLoginCompleted?: () => void;
+    // Represents the screen to display as a result of parsing the initial window.location
+    initialScreenAfterLogin?: IScreen;
+    // displayname, if any, to set on the device when logging in/registering.
+    defaultDeviceDisplayName?: string,
+    // A function that makes a registration URL
+    makeRegistrationUrl: (object) => string,
+}
+
+interface IState {
+    // the master view we are showing.
+    view: VIEWS;
+    // What the LoggedInView would be showing if visible
+    page_type?: PageTypes;
+    // The ID of the room we're viewing. This is either populated directly
+    // in the case where we view a room by ID or by RoomView when it resolves
+    // what ID an alias points at.
+    currentRoomId?: string;
+    currentGroupId?: string;
+    currentGroupIsNew?: boolean;
+    // If we're trying to just view a user ID (i.e. /user URL), this is it
+    currentUserId?: string;
+    // this is persisted as mx_lhs_size, loaded in LoggedInView
+    collapseLhs: boolean;
+    leftDisabled: boolean;
+    middleDisabled: boolean;
+    // the right panel's disabled state is tracked in its store.
+    version?: string;
+    newVersion?: string;
+    hasNewVersion: boolean;
+    newVersionReleaseNotes?: string;
+    checkingForUpdate?: string; // updateCheckStatusEnum
+    showCookieBar: boolean;
+    // Parameters used in the registration dance with the IS
+    register_client_secret?: string;
+    register_session_id?: string;
+    register_id_sid?: string;
+    // When showing Modal dialogs we need to set aria-hidden on the root app element
+    // and disable it when there are no dialogs
+    hideToSRUsers: boolean;
+    syncError?: Error;
+    resizeNotifier: ResizeNotifier;
+    showNotifierToolbar: boolean;
+    serverConfig?: ValidatedServerConfig;
+    ready: boolean;
+    thirdPartyInvite?: object;
+    roomOobData?: object;
+    viaServers?: string[];
+    pendingInitialSync?: boolean;
+}
+
+export default class MatrixChat extends React.PureComponent<IProps, IState> {
+    static VIEWS = VIEWS; // we export this so that the integration tests can use it :-S
```

Comment:
ewwwwwwwwwwwwwwwwwwwwwww

but also fine for now. Can we file a ticket and link to it here?